subsys/mgmt/mcumgr/lib/smp/src/smp.c

```diff
@@ -18,33 +20,41 @@
 #include "smp/smp.h"
 #include "../../../smp_internal.h"
 
-static void
-cbor_nb_reader_init(struct cbor_nb_reader *cnr,
-		    struct net_buf *nb)
+static void cbor_nb_reader_init(struct cbor_nb_reader *cnr, struct net_buf *nb)
 {
-	/* Skip the mgmt_hdr */
-	void *new_ptr = net_buf_pull(nb, sizeof(struct mgmt_hdr));
+	/* Skip the smp_hdr */
+	void *new_ptr = net_buf_pull(nb, sizeof(struct smp_hdr));
 
 	cnr->nb = nb;
 	zcbor_new_decode_state(cnr->zs, ARRAY_SIZE(cnr->zs), new_ptr,
 			       cnr->nb->len, 1);
 }
 
-static void
-cbor_nb_writer_init(struct cbor_nb_writer *cnw, struct net_buf *nb)
+static void cbor_nb_writer_init(struct cbor_nb_writer *cnw, struct net_buf *nb)
 {
 	net_buf_reset(nb);
 	cnw->nb = nb;
-	cnw->nb->len = sizeof(struct mgmt_hdr);
-	zcbor_new_encode_state(cnw->zs, 2, nb->data + sizeof(struct mgmt_hdr),
+	cnw->nb->len = sizeof(struct smp_hdr);
+	zcbor_new_encode_state(cnw->zs, 2, nb->data + sizeof(struct smp_hdr),
 			       net_buf_tailroom(nb), 0);
 }
 
+static void smp_hdr_ntoh(struct smp_hdr *hdr)
+{
+	hdr->nh_len = sys_be16_to_cpu(hdr->nh_len);
+	hdr->nh_group = sys_be16_to_cpu(hdr->nh_group);
+}
+
+static void smp_hdr_hton(struct smp_hdr *hdr)
```

Comment:
This is used for tests and cannot be made static: https://github.com/zephyrproject-rtos/zephyr/pull/44124/files#diff-13980dedd3fed9aaa5b48c06295ce8ecc9c7fa3968fd4322d6212435b3ca0f9aR24

Reply:
This is internal, auxiliary,  function that should have never been exported. The test implements SMP client and can spare some lines (2) to implement the feature.